resolution/controllers/hub_key_handler.py

```diff
@@ -167,7 +164,10 @@ def resolve_link_id_type(reference_links, parsed_key):
     link_for_id_type = None
```

Comment:
1. should this function be in controller?
2. should this function be so complex?